pennylane/estimator/ops/op_math/controlled_ops.py

```diff
@@ -0,0 +1,2174 @@
+        return [GateCount(tof), GateCount(cnot, 2)]
+
+    @classmethod
+    def adjoint_resource_decomp(cls, target_resource_params: dict | None = None) -> list[GateCount]:
+        r"""Returns a list representing the resources for the adjoint of the operator.
+
+        Args:
+            target_resource_params (dict | None): A dictionary containing the resource parameters
+                of the target operator.
+
+        Resources:
+            This operation is self-adjoint, so the resources of the adjoint operation results
+            are same as the originial operation.
+
+        Returns:
+            list[:class:`~.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects,
+            where each object represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        return [GateCount(cls.resource_rep())]
+
+    @classmethod
+    def controlled_resource_decomp(
+        cls, num_ctrl_wires: int, num_zero_ctrl: int, target_resource_params: dict | None = None
+    ) -> list[GateCount]:
+        r"""Returns a list representing the resources for a controlled version of the operator.
+
+        Args:
+            num_ctrl_wires (int): the number of qubits the operation is controlled on
+            num_zero_ctrl (int): the number of control qubits, that are controlled when in the :math:`|0\rangle` state
+            target_resource_params (dict | None): A dictionary containing the resource parameters
+                of the target operator.
+
+        raises:
+            :class:`~.pennylane.exceptions.ResourcesUndefinedError`: Controlled version of this gate is not defined.
+        """
+        raise ResourcesUndefinedError
+
+    @classmethod
+    def pow_resource_decomp(
+        cls, pow_z: int, target_resource_params: dict | None = None
+    ) -> list[GateCount]:
+        r"""Returns a list representing the resources for an operator raised to a power.
+
+        Args:
+            pow_z (int): the power that the operator is being raised to
+            target_resource_params (dict | None): A dictionary containing the resource parameters
+                of the target operator.
+
+        Resources:
+            This operation is self-inverse, thus when raised to even integer powers acts like
+            the identity operator and raised to odd powers it produces itself.
+
+        Returns:
+            list[:class:`~.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects,
+            where each object represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        return (
+            [GateCount(resource_rep(Identity))]
+            if pow_z % 2 == 0
+            else [GateCount(cls.resource_rep())]
+        )
+
+
+class CCZ(ResourceOperator):
+    r"""Resource class for the CCZ gate.
+
+    Args:
+        wires (Sequence[int] | None): the subsystem the gate acts on
```

Comment:
I notice that we say "the wire the operation acts on" for the other ones. Should we be more consistent?
```suggestion
        wires (Sequence[int] | None): the wire the operation acts on
```